src/protocol/serializers/error-serializer.js

```diff
@@ -4,10 +4,33 @@ const errorSerializer = {
         return value instanceof Error;
     },
     serialize(value) {
-        return value.message;
+        var result = {
+            message: value.message
+        };
+
+        if (value.info) {
+            result.info = value.info;
+        }
+
+        if (value.code) {
+            result.code = value.code;
+        }
+
+        return JSON.stringify(result);
```

Comment:
`JSON.stringify` blows away `undefined`s so in theory you could just `return JSON.stringify({message: value.message, code: value.code, info: value.info})`.